SIL.Core.Desktop/Progress/Commands/AsyncCommand.cs

```diff
@@ -146,9 +147,9 @@ public bool Canceling
 		/// Override this method to invoke the actual
 		/// long operation using your preferred async mode.
 		/// </summary>
-		protected abstract void BeginInvokeCore();
+		protected abstract Task BeginInvokeCore();
```

Comment:
Shouldn't this have `async`?

```suggestion
		protected abstract async Task BeginInvokeCore();
```

Reply:
It only needs async if there's an await command within. Since this abstract, there are no await's. And async doesn't modify the method signature, so implementing methods are free to choose whether or not they want to go asynchronous. Additionally, any method returning Task can be awaited -- no async keyword required :)